index.js

```diff
@@ -78,38 +70,42 @@ module.exports = class DfinityTx extends Message {
 
   /**
    * deserializes the message and returns a new instance of `DfinityTx`
-   * @param {Buffer} raw - the serialized raw messsage
+   * @param {Buffer} raw - the serialized raw message
    * @return {Promise} resolve with a new instance of `DfinityTx`
    */
   static async deserialize (raw) {
-    const p = new Pipe(raw)
-    const type = p.read(1)
-    assert.equal(type[0], 1, 'txs should start with type 1')
+    const isSigned = raw.length > 65
+    const rawData = isSigned ? raw.subarray(0, -65) : raw
+    const parts = cbor.decodeAllSync(rawData)
+    const type = parts[0]
```

Comment:
use CBOR's tags instead of type here. I would suggest number 39 or 44 since we are using the following tags: actor is 40, id 41, link 42, function 43